integration-cli/docker_cli_pull_test.go

```diff
@@ -272,6 +272,6 @@ func (s *DockerSuite) TestPullLinuxImageFailsOnWindows(c *check.C) {
 // Regression test for https://github.com/docker/docker/issues/28892
 func (s *DockerSuite) TestPullWindowsImageFailsOnLinux(c *check.C) {
 	testRequires(c, DaemonIsLinux, Network)
-	_, _, err := dockerCmdWithError("pull", "microsoft/nanoserver")
-	assert.ErrorContains(c, err, "cannot be used on this platform")
+	_, _, err := dockerCmdWithError("pull", "mcr.microsoft.com/windows/servercore:ltsc2019")
+	assert.ErrorContains(c, err, "no matching manifest")
```

Comment:
Shouldn't this produce the other error though? https://github.com/moby/moby/blob/fc01c2b481097a6057bec3cd1ab2d7b4488c50c4/distribution/config.go#L154-L160

Reply:
It actually shows this error when the image in the registry is a manifest list.

https://github.com/moby/moby/blob/50e63adf30d33fc1547527a4097c796cbe4b770f/distribution/pull_v2.go#L759

Reply:
Hm... looks like a bug in that detection then; wondering: could we fix that?
Was the previous image not multi-arch?

Reply:
Actually; for this test we're only checking that the pull fails, so I guess it's ok (we might want to check for `no matching manifest for linux` perhaps?)

Reply:
Excellent idea so we don't need a new test for #38574  :-)
I'll add it, but also the `no matching manifest for windows` in the corresponding test for Windows.